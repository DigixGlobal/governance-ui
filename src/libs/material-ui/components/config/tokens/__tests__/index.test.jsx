import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tokens from '../index';

jest.mock('react-redux', () => ({
  connect: () => component => component,
}));

const tokens = [{
  id: 1,
}, {
  id: 2,
}];

describe('<Tokens />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <Tokens
        tokens={tokens}
        createToken={jest.fn}
        updateToken={jest.fn}
        deleteToken={jest.fn}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
