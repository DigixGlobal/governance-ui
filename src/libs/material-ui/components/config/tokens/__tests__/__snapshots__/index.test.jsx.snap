// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tokens /> renders correctly 1`] = `
<Grid>
  <GridColumn
    width={7}
  >
    <Header>
      Tokens
      <HeaderSubheader>
        Configure Available Token Assets
      </HeaderSubheader>
    </Header>
  </GridColumn>
  <GridColumn
    textAlign="right"
    width={9}
  >
    <EZModal
      content={[Function]}
      handleSubmit={[Function]}
      header="Add Token"
      trigger={
        <Button
          as="button"
          basic={true}
          content="Create"
          icon="plus"
        />
      }
    />
  </GridColumn>
  <GridColumn
    width={16}
  >
    <Table
      as="table"
      selectable={true}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <TableBody
        as="tbody"
      >
        <Token
          deleteToken={[Function]}
          key="1"
          token={
            Object {
              "id": 1,
            }
          }
          updateToken={[Function]}
        />
        <Token
          deleteToken={[Function]}
          key="2"
          token={
            Object {
              "id": 2,
            }
          }
          updateToken={[Function]}
        />
      </TableBody>
    </Table>
  </GridColumn>
</Grid>
`;
